flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaConnectorOptions.java

```diff
@@ -248,6 +248,16 @@
                             "If the delivery guarantee is configured as "
                                     + DeliveryGuarantee.EXACTLY_ONCE
                                     + " this value is used a prefix for the identifier of all opened Kafka transactions.");
+    // --------------------------------------------------------------------------------------------
+    // Source specific options
+    // --------------------------------------------------------------------------------------------
+
+    public static final ConfigOption<Long> SOURCE_BOUNDED =
```

Comment:
`Source` interface has the `getBoundedness` public API and thus this option could be the common table option for all connectors. And this option should be the end offsets not boolean type value in your implementation.